### PR TITLE
UI and Functionality to Publish

### LINKED ISSUE #5: Publish project to GitHub
When a project is on a local git repository but has no remote set up, we need to have a `Publish` UI to allow publishing repositories to a GitHub server without having to leave Unity.

- [ ] Detect if user is authenticated to control presenting "Publish" button
- [ ] Detect whether repository has been published (does it have a remote?)
- [ ] Hide Push/Pull buttons and show a Publish button (history view? branches view? everywhere?)
- [ ] Have a publish window that gathers the basic information for publishing a repo
  - Owner (either org or user), Name, Description (optional)
  - Make the repo public or private?

Validations:
- Is the user logged in? If not, we'll need to fire up the authentication flow before being able to publish
- Repository name
  - [ ] Name duplication - we can check whether the wanted name already exists with octokit
  - [ ] Spaces should be turned into dashes
- Private repository
  - [ ] If the user can't create private repositories, we need to show a useful error message

Advanced:
- [ ] Add support for selecting a license

UI Questions:
- The header should reflect whether the repo is published or not. Should we show the url there if it is?

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fixes #5

Replaces #10 

Depends on:
- [x] #109
- [x] #110
- [x] #111

### CODE REVIEW

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        private string repoDescription = "";
+        [SerializeField]
+        private bool togglePrivate;
+
+        [NonSerialized]
+        private IApiClient client;
+        [NonSerialized]
+        private bool isBusy;
+        [NonSerialized]
+        private string error;
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs]: Could you put this stuff all in one line for readibility? Yeah, I know, R# refuses to do it properly, it's a pain in the ass

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+                }
+                GUILayout.EndVertical();
+            }
+            GUILayout.EndHorizontal();
+
+            GUILayout.Label("Description");
+            GUI.enabled = !isBusy;
+            repoDescription = EditorGUILayout.TextField(repoDescription);
+            GUI.enabled = true;
+            GUILayout.Space(5);
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs]: Padding/margin spacing should be set in `const` fields in the Styles so we don't have to track them down everywhere in code. There's a bunch throughout the UI here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            if (error != null)
+                GUILayout.Label(error, Styles.ErrorLabel);
+
+            GUILayout.FlexibleSpace();
+
+            GUILayout.BeginHorizontal();
+            {
+                GUILayout.FlexibleSpace();
+                GUI.enabled = !string.IsNullOrEmpty(repoName) && !isBusy;
+                if (GUILayout.Button("Create"))
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs]: Button strings should be in `const` fields, we'll need to localize them at some point and it's easier if they're not in code.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs (hunk 1 above)
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs]: no problem 👍

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs (hunk 2 above)
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs]: cool 👍

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs (hunk 3 above)
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PUblishView.cs]: got it 👍

**Pull request merged**